app/back_end/data/refactoring.py

```diff
@@ -10,38 +10,53 @@
+    """
+    Convert a Pandas Series to integer or numeric, returning the original series on failure.
+    This function first tries to convert the Series to integer.
+    If that fails, it attempts to convert it to numeric (float).
+    Args:
+        series (pd.Series): The Pandas Series to be converted to numeric types.
+    Returns:
+        pd.Series: A Series converted to integer or numeric type, or the original Series
+        if conversion is unsuccessful.
+    """
+    try:
+        return  pd.to_numeric(series,downcast="integer")
+    except (ValueError, TypeError):
+        try:
+            return pd.to_numeric(series)  # Fallback to float
+        except (ValueError, TypeError):
+            return series
+
 def set_lovd_dtypes(df_dict):
     """
     Convert data from LOVD format table to desired data format based on specified data types.
 
-    :param dict[str, tuple[DataFrame, list[str]] df_dict: Dictionary of tables saved as DataFrame
+    :param dict[str, DataFrame] df_dict: Dictionary of tables saved as DataFrame
     """
 
-    for table_name in df_dict:
-        frame: DataFrame = df_dict[table_name]
+    for table_name, frame in df_dict.items():
+        frame = frame.convert_dtypes()
         for column in frame.columns:
-            if column not in LOVD_TABLES_DATA_TYPES[table_name]:
-                raise ValueError(f"Column {column} is undefined in LOVD_TABLES_DATA_TYPES")
-
-            match LOVD_TABLES_DATA_TYPES[table_name][column]:
-                case "Date":
+            frame[column] = safe_to_numeric(frame[column])
+            match frame[column].dtype:
+                case pd.Int64Dtype() | pd.Int32Dtype() | pd.Int16Dtype() | pd.Int8Dtype():
+                    frame[column] = frame[column].astype('Int64')
+                case pd.Float64Dtype():
+                    frame[column] = frame[column].astype('float64')
+                case pd.BooleanDtype():
+                    frame[column] = frame[column].map({"0": False, "1": True}).astype('boolean')
+                case _ if pd.api.types.is_datetime64_any_dtype(frame[column]):
                     frame[column] = pd.to_datetime(frame[column], errors='coerce')
-                case "Boolean":
-                    frame[column] = frame[column].map({"0": False, "1": True})
-                case "String":
+                case pd.StringDtype():
                     frame[column] = frame[column].astype('string')
-                case "Integer":
-                    frame[column] = pd.to_numeric(frame[column]).astype('Int64')
-                case "Double":
-                    frame[column] = pd.to_numeric(frame[column]).astype('float')
                 case _:
-                    raise ValueError(f"Undefined data type: "
-                                     f"{LOVD_TABLES_DATA_TYPES[table_name][column]}")
-
+                    raise ValueError(f"Undefined data type for column '{column}': {frame[column].dtype}")
+        df_dict[table_name] = frame
```

Comment:
You already converted with using `convert_dtypes`. Why do you continue to convert?

Reply:
Because it automatically just converts everything to string